### ISSUE TITLE
use getrandom() on Linux

### ISSUE THREAD

**Issue by wilzbach** [libmir/mir-random#10]
Since Linux 3.17 there is a `getrandom` syscall - it's pretty cool because it always to block until enough entropy is provided see e.g.

https://lwn.net/Articles/606141/

or:

http://man7.org/linux/man-pages/man2/getrandom.2.html

Small proof-of-concept for Linux:

```d
/*
 * Flags for getrandom(2)
 *
 * GRND_NONBLOCK	Don't block and return EAGAIN instead
 * GRND_RANDOM		Use the /dev/random pool instead of /dev/urandom
 */
enum GRND_NONBLOCK = 0x0001;
enum GRND_RANDOM = 0x0002;
enum GETRANDOM = 318;

size_t syscall(size_t ident, size_t n, size_t arg1, size_t arg2)
{
    size_t ret;

    synchronized asm
    {
        mov RAX, ident;
        mov RDI, n[RBP];
        mov RSI, arg1[RBP];
        mov RDX, arg2[RBP];
        syscall;
        mov ret, RAX;
    }
    return ret;
}

void main() {
    long buf;
    syscall(GETRANDOM, cast(size_t) &buf, buf.sizeof, 0);

    import std.stdio;
    writeln(buf);
}
```

syscall is taken from [syscall-d](https://github.com/kubo39/syscall-d).

Also should we provide a way for the user to directly receive random data via OS APIs?

**Comment by 9il** [comment]
> Also should we provide a way for the user to directly receive random data via OS APIs?

Yes

> Since Linux 3.17 there is a getrandom syscall - it's pretty cool because it always to block until enough entropy is provided see e.g.

Awesome! Why not to use extern(C) getrandom?

Looking forward for the PR!

**Comment by 9il** [comment]
Fixed in v0.2.5
